woocommerce/woocommerce-android
# Get gift card data

⚠️ This PR depends on this tiny [FluxC PR](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2696)

Closes: #8398

### Why
We are adding gift card read-only support to the Woo App

### Description
This PR adds the necessary changes _to fetch the gift card information from the API_ . Actually, it doesn't bring the whole gift card information from the API, but relies only on the summary gift card information the orders request contains. It also adds a minor change in the decision-making logic of whether fetch hosted woo extension information or not. Now the extension should appear installed and active in the local database to fetch the data.

### Testing instructions
Prerequisites:
* Install the gift card extension in your store and place some orders where you apply/redeem some gift cards

TC1 
1. Open the orders tab
2. Select an order with a gift card applied/redeemed
3. Check that the gift card information appears on the order details (The Gift Card section and the Payment Section)

TC2
1. Open the orders tab
2. Select an order where no gift card was applied/redeemed
3.  Check that NO gift card information appears

TC3
1. Deactivate the Gift Card extension
2. Close/Open the app (to refresh the list of plugins)
3. Open the orders tab
4. Select an order with a gift card applied/redeemed
5. Check that NO gift card information appears (because the plugin is deactivated)

### Images/gif

https://user-images.githubusercontent.com/18119390/229257516-9ba76efa-8d31-4f9f-86a3-c79d66a85437.mp4
